src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package mate.academy.security.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.DataValidationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.security.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        isValidEmail(email);
+        isValidPassword(password);
+        if (userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException("Can't register user");
+        }
+        User user = new User(email, password);
+        return userService.add(user);
+    }
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User user = userService.findByEmail(email)
+                .orElseThrow(() -> new AuthenticationException("Can't authenticate user: "
+                        + "Wrong email " + email));
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (!user.getPassword().equals(hashedPassword)) {
+            throw new AuthenticationException("Can't authenticate user: "
+                    + "Wrong password " + password);
+        }
```

Comment:
![image](https://user-images.githubusercontent.com/49767049/230957315-2498f00a-afe6-4c3a-92f5-ef289e7d9963.png)
throw exception only once, with standard message - Email/password is incorrect. Don't use orElseThrow here. You can get optional and then use isEmpty(), get() for example